sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/query/ContinuationResumeLogicTests.java

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.query;
+
+import com.azure.cosmos.implementation.Strings;
+import com.azure.cosmos.implementation.feedranges.FeedRangeEpkImpl;
+import com.azure.cosmos.implementation.routing.Range;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ContinuationResumeLogicTests {
```

Comment:
>ContinuationResumeLogicTests [](start = 13, length = 28)

It's out of the scope of this PR, but in the future I think we will need to port the Pagination library, so that we can end to end test splits/merges in memory. You can also validate against a prod account for now to do a manual validation.

Reply:
Yes, I already have a prod test that tests the split scenario for continuation token